cost-model/src/transaction_cost.rs

```diff
@@ -1,10 +1,109 @@
-use solana_sdk::pubkey::Pubkey;
+use {crate::block_cost_limits, solana_sdk::pubkey::Pubkey};
+
+/// TransactionCost is used to represent resources required to process
+/// a transaction, denominated in CU (eg. Compute Units).
+/// Resources required to process a regular transaction often include
+/// an array of variables, such as execution cost, loaded bytes, write
+/// lock and read lock etc.
+/// Vote has a simpler and pre-determined format, it's cost structure
+/// can be simpler, calculation quicker.
+#[derive(Debug)]
+pub enum TransactionCost {
+    Vote { writable_accounts: Vec<Pubkey> },
+    Transaction(UsageCostDetails),
+}
+
+impl TransactionCost {
+    pub fn sum(&self) -> u64 {
+        match self {
+            Self::Vote { writable_accounts } => {
+                let num_of_signatures = writable_accounts.len() as u64;
+
+                solana_vote_program::vote_processor::DEFAULT_COMPUTE_UNITS
+                    .saturating_add(
+                        block_cost_limits::SIGNATURE_COST.saturating_mul(num_of_signatures),
+                    )
+                    .saturating_add(
+                        block_cost_limits::WRITE_LOCK_UNITS.saturating_mul(num_of_signatures),
+                    )
```

Comment:
Technically, to follow cost model, the usage cost of a simple vote transaction should the sum of: 
1. the static `vote` program cost,
2. 1 or 2 signature cost, 
3. 1 or 2 write lock cost,
4. a small cost for instruction data (bytes of `Vec<Slot>` it votes on), normal 20-ish CU
5. the write locked accounts' loaded size cost.

First 3 are captured here, the 4th is small enough can be ignored, the last one is an unknown. I'm going to check what are the those accounts look like in mainnet-beta, hoping they all have similar and stable size, allowing to assign a static value to it. 

Regardless, should we worry if a vote tx loads a huge account to slow down replay?

Reply:
looks like it is plausible to set static cost for vote: #33269 

If ended up actually doing it, perhaps need to make sure vote violate accounts assumption (all fixed size) should fail asap.